backend/controller/controller.go

```diff
@@ -1382,6 +1390,26 @@ func (s *Service) executeAsyncCalls(ctx context.Context) (time.Duration, error)
 		observability.AsyncCalls.Acquired(ctx, call.Verb, call.Origin.String(), call.ScheduledAt, err)
 		return 0, err
 	}
+
+	// Extract the otel context from the call
+	var oc propagation.MapCarrier
+	err = json.Unmarshal(call.TraceContext, &oc)
+	if err != nil {
+		observability.AsyncCalls.Acquired(ctx, call.Verb, call.Origin.String(), call.ScheduledAt, err)
+		return 0, fmt.Errorf("failed to unmarshal otel context: %w", err)
+	}
+	ctx = otel.GetTextMapPropagator().Extract(ctx, oc)
+
+	// Extract the request key from the call and attach it as the parent request key
+	parentRequestKey := optional.None[model.RequestKey]()
+	if prk, ok := call.ParentRequestKey.Get(); ok {
+		if rk, err := model.ParseRequestKey(prk); err == nil {
+			parentRequestKey = optional.Some(rk)
+		} else {
+			logger.Tracef("Ignoring invalid request key: %s", prk)
+		}
+	}
+
```

Comment:
nit: Would it make sense to have this in `observability/async_calls` to keep it out of the controller? That would also remove the `otel` dependencies here.